### ISSUE TITLE
When `nil` is coerced into `RArray`, it becomes `[nil]`

### ISSUE THREAD

**Issue by gjtorikian** [matsadler/magnus#38]
Unlike https://github.com/matsadler/magnus/issues/35, I don't have time to set up a test repro at the moment, but hopefully this example will suffice. Let me know if you'd like to see a demo program and I can craft one.

Basically, if I do this in Ruby:

```ruby
SomeClass.new(var: nil)
```

And in Magnus, do:

```rust
        let args = scan_args::scan_args(args)?;
        let _: () = args.required;
        let _: () = args.optional;
        let _: () = args.splat;
        let _: () = args.trailing;
        let _: () = args.block;

        let kw = scan_args::get_kwargs::<
            _,
            (),
            (
                Option<RArray>,
            ),
            (),
        >(args.keywords, &[], &["var"])?;
        let (rb_var) = kw.optional;
```

According to `rb_var.inspect()`, this is `[nil]`, not `nil`; and thus, a `TypeError` is never raised, although I believe it should.

**Comment by matsadler** [comment]
So Ruby's C API has a bunch of functions to check a value is of a certain type (or convert it to that type with the implicit conversion methods, `to_str`, `to_ary`, etc).

There's some like `rb_check_<type>_type` that return the value if it's of that type, call `to_<typ>` if it responds to that method, or return `nil` if it can't be converted. Magnus looks for that `nil` and raises an error.

Then there's `rb_str_to_str` which returns the value if it's a string, calls `to_str` if the object responds to it, or *errors* if it can't be converted. This is convenient as then the code in Magnus doesn't have to do the extra `nil` check to return an error in the last case.

The array conversions are done using `rb_ary_to_ary`. Turns out this does not work like `rb_str_to_str`. Instead of erroring if the value can't be converted (or returning `nil`) it returns the value wrapped in an array.

In my defence, at the time I wrote this code the Ruby C API was totally undocumented (the headers now do document this potential confusion).

I've fixed this in 69ff299, but as it's a change in behaviour I'm going to put of releasing till 0.5.

**Comment by gjtorikian** [comment]
> In my defence, at the time I wrote this code the Ruby C API was totally undocumented (the headers now do document this potential confusion).

No worries at all! Thank you for the project, I’m just calling what I see. 😄 Thanks for the quick fix!
